configdialog.cpp

```diff
@@ -518,6 +535,271 @@ void PaletteCustomizeDialog::get_and_save_color(int palette_index) {
+					for (wstring color : colors[i]) {
+						if (color.find_first_of(L';') == wstring::npos) {
+							count++;
+						}
+					}
+					error_info += L" Too few colors defined (have " + to_wstring(count) + L", need " + to_wstring(colors_amount) + L"),";
+				}
+				if ((is_valid[i] & static_cast<uint8_t>(Errors::colors_incorrect_length)) != static_cast<uint8_t>(Errors::none)) {
+					error_info += L" One or more colors were of incorrect length (found:";
+					for (wstring color : colors[i]) {
+						if (color.find_first_of(';') == wstring::npos) {
+							error_info += L" " + to_wstring(color.length()) + L",";
+						}
+					}
+					*error_info.rbegin() = L';';
+					error_info += L" expected: 6),";
+				}
+				if ((is_valid[i] & static_cast<uint8_t>(Errors::colors_invalid_characters)) != static_cast<uint8_t>(Errors::none)) {
+					wstring invalid_chars = L"";
+					for (wstring color : colors[i]) {
+						size_t index = 0;
+						while (color.find_first_not_of(valid_hex_chars + L';', index) != wstring::npos) {
+							index = color.find_first_not_of(valid_hex_chars + L';', index);
+							if (invalid_chars.find_first_of(color[index]) == wstring::npos) {
+								invalid_chars += color[index];
+							}
+							index++;
+						}
+					}
+					error_info += L" Colors have invalid characters (" + invalid_chars + L"),";
+				}
+
+				if (invalid < count_invalid) {
+					*error_info.rbegin() = L';';
+					error_info += L"\n";
+				}
+				else {
+					error_info.pop_back();
+				}
+
+				if (invalid >= 10 && invalid < count_invalid) {
+					error_info += L"... + " + to_wstring(count_invalid - invalid) + L" more palettes";
+					break;
+				}
+			}
+		}
+		
+
+		if (count_invalid >= palettes_amount) {
+			error_info += L"\n\nNo palettes can be imported.";
+			MessageBox(
+				NULL,
+				error_info.c_str(),
+				L"Palette Customizer",
+				MB_OK | MB_ICONERROR
+			);
+			import = false;
+		}
+		else {
+			error_info += L"\n\nWould you like to import the remaining palettes anyway?\n\nThe following palettes can be imported:";
+			for (size_t i = 0; i < palettes_amount; i++) {
+				if (is_valid[i] == static_cast<uint8_t>(Errors::none)) {
+					error_info += L" " + names[i] + L",";
+				}
+			}
+			*error_info.rbegin() = L'.';
+			int num = MessageBox(
+				NULL,
+				error_info.c_str(),
+				L"Palette Customizer",
+				MB_YESNO | MB_ICONERROR | MB_DEFBUTTON2
+			);
+			import = (num == IDYES);
+		}
+	}
+	else {
+		wstring import_confirmation = L"The following palettes will be imported:";
+		for (size_t i = 0; i < palettes_amount; i++) {
+			if (is_valid[i] == static_cast<uint8_t>(Errors::none)) {
+				import_confirmation += L" " + names[i] + L",";
+			}
+		}
+		*import_confirmation.rbegin() = L'.';
+		import_confirmation += L"\n\nConfirm palette import?";
+		int num = MessageBox(
+			NULL,
+			import_confirmation.c_str(),
+			L"Palette Customizer",
+			MB_YESNO | MB_ICONINFORMATION
+		);
+		import = (num == IDYES);
+	}
+
+	if (import) {
```

Comment:
Please for the love of God make this a separate function. One function for validating, one for actually importing. This can't possibly get any longer.